webapp/mixins/scrollToAnchor.js

```diff
@@ -0,0 +1,23 @@
+function scrollToAnchor(anchor, { checkAnchor, $scrollTo }) {
+  if (typeof checkAnchor !== 'function')
+    throw new Error(
+      'You must define `checkAnchor` on the component if you use scrollToAnchor mixin!',
+    )
+  if (!checkAnchor(anchor)) return
+  setTimeout(() => {
+    $scrollTo(anchor)
+  }, 250)
+}
+
+export default {
+  watch: {
+    $route(to, from) {
+      const anchor = to && to.hash
```

Comment:
> [<img alt="vbelolapotkov" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/vbelolapotkov) **Authored by [vbelolapotkov](https://github.com/vbelolapotkov)**
_<time datetime="2019-10-01T10:45:28Z" title="Tuesday, October 1st 2019, 12:45:28 pm +02:00">Oct 1, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,20 @@
+import { scrollTo } from 'vue-scrollto'
+
+export default {
+  watch: {
+    $route(to, from) {
+      const anchor = to && to.hash
+      if (!this.checkAnchor(anchor)) return
+      setTimeout(() => {
+        scrollTo(anchor, 1000)
+      }, 250)
+    },
+  },
+  mounted() {
+    const anchor = this.$route && this.$route.hash
+    if (!this.checkAnchor(anchor)) return
```


@roschaefer how about some default behaviour when `checkAnchor` method is not defined in the component?